### ISSUE TITLE
Error in sklearn$metrics$precision_score(y_test, y_pred, average = "weighted") :    attempt to apply non-function

### ISSUE THREAD

**Issue by ZijiaLai** [dzhakparov/GeneSelectR#26]
`> load("UrbanRandomSubset.rda")
> View(UrbanRandomSubset)
> head(UrbanRandomSubset[, 1:10])  # Display the first 10 columns of the first few rows
        treatment ENSG00000174371__EXO1 ENSG00000123600__METTL8 ENSG00000154124__OTULIN
CA002YF  Urban_AD          2.984508e+00            1.816292e+00            4.959833e+00
CA009ST  Urban_AD           0.565690377             2.010920739             4.601555569
CA010EB  Urban_AD          3.6708983671            3.1615995794            4.6202078459
CA011LQ  Urban_AD          1.408499e+00            2.674512e+00            3.909325e+00
CA014LB  Urban_AD          2.5183409223            1.8185148529            3.8273512783
CA015AM  Urban_AD          2.6664729139            3.0494281366            4.4946295998
        ENSG00000006607__FARP2 ENSG00000135686__KLHL36 ENSG00000130348__QRSL1
CA002YF           4.125659e+00            6.584945e+00           4.298858e+00
CA009ST            4.873391176             6.677867796            4.043031406
CA010EB           4.8954199262            6.0682238886           4.4417887589
CA011LQ           3.852403e+00            6.041770e+00           4.185302e+00
CA014LB           4.6382172341            6.3209867192           4.3729324835
CA015AM           4.4197758295            5.9998300955           4.6819749406
        ENSG00000268041__AC010616.1 ENSG00000163812__ZDHHC3 ENSG00000233041__PHGR1
CA002YF                3.150753e+00            4.617841e+00          -2.934457e+00
CA009ST                 3.090960290             5.161640677           -2.934457404
CA010EB                3.3511595489            4.9131957592          -2.9344574043
CA011LQ                3.049633e+00            5.061451e+00          -2.318961e+00
CA014LB                2.6071577494            5.1488088821          -2.9344574043
CA015AM                2.6885696358            5.0960550151          -2.9344574043
> View(UrbanRandomSubset)
> table(UrbanRandomSubset$treatment)  # Display the distribution of the 'treatment' column

     Urban_AD Urban_Healthy 
           31            29 
> # Prepare the feature matrix and the label vector
> X <- UrbanRandomSubset %>% dplyr::select(-treatment)  # Select all columns except 'treatment' as features
> y <- UrbanRandomSubset["treatment"]  # Extract the 'treatment' column as the label vector
> y <- as.factor(y[, 1])  # Convert the label vector to a factor
> # Perform feature selection using GeneSelectR
> selection_results <- GeneSelectR(X = X, y = y, njobs = -1,  # Perform feature selection using all available cores
+                                  perform_test_split = TRUE,  # Split the data for testing
+                                  calculate_permutation_importance = TRUE)  # Calculate permutation importance
Performing feature selection procedure. Please wait, it takes some time
$preprocessing
Module(sklearn.preprocessing)

$model_selection
Module(sklearn.model_selection)

$feature_selection
Module(sklearn.feature_selection)

$ensemble
Module(sklearn.ensemble)

$pipeline
<class 'sklearn.pipeline.Pipeline'>
 signature: (steps, memory=None, verbose=False)

$forest
<class 'sklearn.ensemble._forest.RandomForestClassifier'>
 signature: (
   n_estimators=100, 
   criterion='gini', 
   max_depth=None, 
   min_samples_split=2, 
   min_samples_leaf=1, 
   min_weight_fraction_leaf=0.0, 
   max_features='auto', 
   max_leaf_nodes=None, 
   min_impurity_decrease=0.0, 
   min_impurity_split=None, 
   bootstrap=True, 
   oob_score=False, 
   n_jobs=None, 
   random_state=None, 
   verbose=0, 
   warm_start=False, 
   class_weight=None, 
   ccp_alpha=0.0, 
   max_samples=None
)

$boruta
Module(boruta)

$randomized_grid
<class 'sklearn.model_selection._search.RandomizedSearchCV'>
 signature: (
   estimator, 
   param_distributions, 
   n_iter=10, 
   scoring=None, 
   n_jobs=None, 
   iid='deprecated', 
   refit=True, 
   cv=None, 
   verbose=0, 
   pre_dispatch='2*n_jobs', 
   random_state=None, 
   error_score=nan, 
   return_train_score=False
)

$grid
<class 'sklearn.model_selection._search.GridSearchCV'>
 signature: (
   estimator, 
   param_grid, 
   scoring=None, 
   n_jobs=None, 
   iid='deprecated', 
   refit=True, 
   cv=None, 
   verbose=0, 
   pre_dispatch='2*n_jobs', 
   error_score=nan, 
   return_train_score=False
)

$bayesianCV
<class 'skopt.searchcv.BayesSearchCV'>
 signature: (
   estimator, 
   search_spaces, 
   optimizer_kwargs=None, 
   n_iter=50, 
   scoring=None, 
   fit_params=None, 
   n_jobs=1, 
   n_points=1, 
   iid='deprecated', 
   refit=True, 
   cv=None, 
   verbose=0, 
   pre_dispatch='2*n_jobs', 
   random_state=None, 
   error_score='raise', 
   return_train_score=False
)

$lasso
<class 'sklearn.linear_model._logistic.LogisticRegression'>
 signature: (
   penalty='l2', 
   dual=False, 
   tol=0.0001, 
   C=1.0, 
   fit_intercept=True, 
   intercept_scaling=1, 
   class_weight=None, 
   random_state=None, 
   solver='lbfgs', 
   max_iter=100, 
   multi_class='auto', 
   verbose=0, 
   warm_start=False, 
   n_jobs=None, 
   l1_ratio=None
)

$univariate
<class 'sklearn.feature_selection._univariate_selection.GenericUnivariateSelect'>
 signature: (
   score_func=<function f_classif at 0x000001EDC38124C0>, 
   mode='percentile', 
   param=1e-05
)

$select_model
<class 'sklearn.feature_selection._from_model.SelectFromModel'>
 signature: (
   estimator, 
   threshold=None, 
   prefit=False, 
   norm_order=1, 
   max_features=None
)

$GradBoost
<class 'sklearn.ensemble._gb.GradientBoostingClassifier'>
 signature: (
   loss='deviance', 
   learning_rate=0.1, 
   n_estimators=100, 
   subsample=1.0, 
   criterion='friedman_mse', 
   min_samples_split=2, 
   min_samples_leaf=1, 
   min_weight_fraction_leaf=0.0, 
   max_depth=3, 
   min_impurity_decrease=0.0, 
   min_impurity_split=None, 
   init=None, 
   random_state=None, 
   max_features=None, 
   verbose=0, 
   max_leaf_nodes=None, 
   warm_start=False, 
   presort='deprecated', 
   validation_fraction=0.1, 
   n_iter_no_change=None, 
   tol=0.0001, 
   ccp_alpha=0.0
)

<class 'sklearn.feature_selection._univariate_selection.GenericUnivariateSelect'>
 signature: (
   score_func=<function f_classif at 0x000001EDC38124C0>, 
   mode='percentile', 
   param=1e-05
)
Fitting the data split: 1 
Fitting pipeline for Lasso feature selection method
RandomForestClassifier(bootstrap=True, ccp_alpha=0.0, class_weight=None,
                       criterion='gini', max_depth=None, max_features='auto',
                       max_leaf_nodes=None, max_samples=None,
                       min_impurity_decrease=0.0, min_impurity_split=None,
                       min_samples_leaf=1, min_samples_split=2,
                       min_weight_fraction_leaf=0.0, n_estimators=100,
                       n_jobs=None, oob_score=False, random_state=None,
                       verbose=0, warm_start=False)
$classifier__n_estimators
[1] 100 500 900

$classifier__max_depth
[1] 10 20 30 40 50

$classifier__min_samples_split
[1]  2  5 10

$classifier__min_samples_leaf
[1] 1 2 4

$classifier__max_features
[1] "auto" "sqrt"

Fitting 5 folds for each of 10 candidates, totalling 50 fits
[Parallel(n_jobs=-1)]: Using backend LokyBackend with 32 concurrent workers.
[Parallel(n_jobs=-1)]: Done  38 out of  50 | elapsed:    0.6s remaining:    0.1s
[Parallel(n_jobs=-1)]: Done  50 out of  50 | elapsed:    0.9s finished
Error in sklearn$metrics$precision_score(y_test, y_pred, average = "weighted") : 
  attempt to apply non-function`

Hi, there seems to be a problem with the code, it is reporting an error as above when using the sample data.
